### ISSUE TITLE
freeze base model

### ISSUE THREAD

**Issue by woiza** [ThilinaRajapakse/simpletransformers#291]
How to freeze all layers of bert and just train task based classifier?

https://github.com/huggingface/transformers/issues/400

**Comment by ThilinaRajapakse** [comment]
There is no direct support for this at the moment. You can edit the `train()` method of a Model class to freeze the layers if you want though.

**Comment by woiza** [comment]
Got it. Thanks for your answer and this awesome project!

**Comment by sgummidipundi** [comment]
Hello! I wanted to see if this functionality has been implemented. I found the following paragraph on the page:

```
Custom parameter groups (freezing layers)

Simple Transformers supports custom parameter groups which can be used to set different learning rates for different layers in a model, freeze layers, train only the final layer, etc.
```

Is there an example that shows how you can use this to freeze the base layers but only train the last classification layer (in the case of classification)?

Thanks

**Comment by ThilinaRajapakse** [comment]
There is some more information in the [docs](https://simpletransformers.ai/docs/usage/#custom-parameter-groups-freezing-layers).

There is also this Medium [article](https://towardsdatascience.com/hyperparameter-optimization-for-optimum-transformer-models-b95a32b70949?source=friends_link&sk=7d19ce15c9ac1230642d826b9deeb638).

**Comment by sgummidipundi** [comment]
I see that there are options to set things like differential learning rates for each of the layers separately. Do you mean to say I would do something like set the learning rate for all of them to zero or something? Or do you mean that there is an option to specifically freeze all the layers?

**Comment by ThilinaRajapakse** [comment]
_The train_custom_parameters_only option is used to facilitate the training of specific parameters only. If train_custom_parameters_only is set to True, only the parameters specified in either custom_parameter_groups or in custom_layer_parameters will be trained._

So, specify the layers/parameters you want to train, and set `train_custom_parameters_only=True` in `model_args`.

**Comment by sgummidipundi** [comment]
If I were to leave out specific layers or groups, then that would mean they would not be trained at all? Is there a way for me to specify just the classification head? How would I be able to find which layer that is?

**Comment by ThilinaRajapakse** [comment]
`print(model.get_named_parameters())`

Anything starting with `classifier.` is part of the classification head.

**Comment by Onkar-2803** [comment]
Any updates on how to freeze particular layers in the model?
I want to freeze some encide layers in NER model.

**Comment by ChristopherMarais** [comment]
Any update on this, it is still not clear.

**Comment by ChristopherMarais** [comment]
Is there a way to keep the learning rate for all the layers the same and only freeze specific layers. I specifically want to do this with a sweep with wandb
